js/article.js

```diff
@@ -1,49 +1,117 @@
+      url: 'data/blogArticles.json',
+      success: function(data, message, xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        if (!localStorage.eTag || eTag !== localStorage.eTag) {
+          Article.getAll(nextFunction); // DONE: pass 'nextFunction' into Article.getAll();
+        } else {
+          Article.loadAll(JSON.parse(localStorage.blogArticles));
+          // Replace the following line with 'nextFunction' and invoke.
+          nextFunction();
+        }
+      }
+    });
+  };
 
-//FUNCTIONALITY TO LOAD ALL ARTICLES IN DECENDING ORDER OF PUBLICATION
-Article.loadAll = function(inputData) {
-  inputData.sort(function(a,b) {
-    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
-  }).forEach(function(ele) {
-    Article.allArticles.push(new Article(ele));
-  });
-};
-
-// THIS ADDS FUNCTIONALITY TO RETRIEVE DATA FROM LOCAL OR REMOTE SOURCE, WHICH PROCESS AND HANDS OFF CONTROL TO THE VIEW
-Article.fetchAll = function() {
-
-  if (localStorage.hackerIpsum) {
-    Article.loadAll(JSON.parse(localStorage.hackerIpsum));
-    articleView.renderIndexPage();
-  } else {
-    $.getJSON('data/hackerIpsum.json', function(data, status, XHR) {
-      localStorage.hackerIpsum = JSON.stringify(data);
-      console.log(data);
-      Article.loadAll(data);
-      articleView.renderIndexPage();
+  Article.getAll = function(nextFunction) {
+    $.getJSON('data/blogArticles.json', function(responseData, message, xhr) {
+      localStorage.eTag = xhr.getResponseHeader('eTag');
+      Article.loadAll(responseData);
+      localStorage.blogArticles = JSON.stringify(responseData);
+      // Invoke our parameter.
+      nextFunction();
     });
   };
-};
+
+  // GETS # OF WORDS IN GIVEN ARTICLE
+  Article.getNumWords = function(article) {
+    return article.body.split(' ').length;
+  };
+
+  Article.numWordsAll = function() {
+    return Article.allArticles.map(function(currentArticle) {
+      return currentArticle.body.match(/\w+/g).length;
+    }).reduce(function(acc, cur) {
+      return acc + cur;
+    });
+  };
+
+  Article.allAuthors = function() {
+      //  returns a mapped collection
+    var authorNames = Article.allArticles.map(function(currentArticle) {
+      return currentArticle.author;
+    }).reduce(function(authorUnique, currentAuthor, index, array) {
+      if(authorUnique.indexOf(currentAuthor) === -1) {
+        authorUnique.push(currentAuthor);
+      }
+      return authorUnique;
+    }, []);
+    return authorNames;
+  };
+
+  Article.numWordsByAuthor = function() {
+    return Article.allAuthors().map(function(currentAuthor) {
+      return {
+        name: currentAuthor,
+        numWords:
+        Article.allArticles.filter(function(currentArticle) {
+          return currentArticle.author === currentAuthor;
+        }).map(function(currentArticle) {
+          return currentArticle.body.match(/\w+/g).length;
+        }).reduce(function(acc, cur) {
+          return acc + cur;
+        })
+      };
```

Comment:
Should have scrolled down before leaving the comment up above. I see now that you are trying to get number of words per author. This is a cool feature but I would consider trying to adjust it to count something different on your site. I had a couple ideas. Since I am doing a portfolio of my past projects I thought about adding a category for skills used. Then you could create a count for how many skills used throughout your portfolio. My other idea was having how many hours I spent on each project and then having a total count of the hours spent on all the projects. Those are just a couple ideas of stats you could include on your portfolio.